core/lib/zksync_core/src/consensus/tests.rs

```diff
@@ -0,0 +1,86 @@
+use super::*;
+use zksync_concurrency::{ctx, scope};
+use zksync_consensus_executor::testonly::FullValidatorConfig;
+use zksync_consensus_roles::validator;
+use zksync_dal::ConnectionPool;
+use zksync_types::{Address, ProtocolVersionId};
+
+fn latest_protocol_version() -> validator::ProtocolVersion {
+    (ProtocolVersionId::latest() as u32)
+        .try_into()
+        .expect("latest protocol version is invalid")
+}
+
+// In the current implementation, consensus certificates are created asynchronously
+// for the miniblocks constructed by the StateKeeper. This means that consensus actor
+// is effectively just backfilling the consensus certificates for the miniblocks in storage.
+#[tokio::test(flavor = "multi_thread")]
+async fn test_backfill() {
+    zksync_concurrency::testonly::abort_on_panic();
+    let ctx = &ctx::test_root(&ctx::RealClock);
+    let rng = &mut ctx.rng();
+    let pool = ConnectionPool::test_pool().await;
+
+    scope::run!(ctx, |ctx, s| async {
+        // Start state keeper.
+        const OPERATOR_ADDRESS: Address = Address::repeat_byte(17);
+        let (mut sk, sk_runner) = testonly::StateKeeperHandle::new(OPERATOR_ADDRESS);
+        s.spawn_bg(sk_runner.run(ctx, &pool));
+
+        // Populate storage with a bunch of blocks.
+        sk.push_random_blocks(rng, 10).await;
+        sk.sync(ctx, &pool).await.context("sk.sync(<1st phase>)")?;
+
+        // Prepare genesis block for consensus.
+        const GENESIS_BLOCK: validator::BlockNumber = validator::BlockNumber(5);
+        let genesis_payload = {
+            let mut storage = storage::storage(ctx, &pool).await.unwrap();
+            storage
+                .fetch_payload(ctx, GENESIS_BLOCK, OPERATOR_ADDRESS)
+                .await
+                .unwrap()
+                .unwrap()
```

Comment:
Dumb question: Can't errors be propagated here?

Reply:
done